pymeasure/instruments/hp/hp8116a.py

```diff
@@ -0,0 +1,541 @@
+        """ Wait until the commands have been processed by the 8116A. """
+        start = time.time()
+        while self.status & Status.buffer_not_empty:
+            time.sleep(0.001)
+            if time.time() - start > timeout:
+                raise RuntimeError('Timeout waiting for commands to be processed.')
+
+    def check_has_option_001(self):
+        """ Return True if the 8116A has option 001 and False otherwise.
+
+        This is done by checking the length of the response to the CST (current status) command
+        which includes sweep parameters and burst parameters only if the 8116A has option 001.
+        """
+
+        # The longest possible state string is 163 characters long including termination characters
+        state_string = self.ask('CST', 163).split('\r\n')[0].strip(' ,\r\n')
+
+        if len(state_string) == 159:
+            return True
+        elif len(state_string) == 87:
+            return False
+        else:
+            log.warning('Could not determine if 8116A has option 001. Assuming it has.')
+            return True
+
+    # Instrument communication #
+
+    def write(self, command):
+        """ Write a command to the instrument and wait until the 8116A has interpreted it. """
+        self.adapter.write(command)
+
+        # We need to read the status byte and wait until the buffer_not_empty bit
+        # is cleared because some older units lock up if we don't.
+        self.wait_for_commands_processed()
+
+    def read(self):
+        """ Some units of the 8116A don't use the EOI line (see service note 8116A-07A).
+        Therefore reads with automatic end-of-transmission detection will timeout.
+        Instead, :code:`adapter.read_bytes()` has to be used.
+        """
+        raise NotImplementedError('Not supported, use adapter.read_bytes() instead')
+
+    def ask(self, command, num_bytes=None):
+        """ Write a command to the instrument, read the response, and return the response as ASCII text.
+
+        :param command: The command to send to the instrument.
+        """
+        self.write(command)
+
+        if num_bytes is None:
+            if command == 'CST':
+                # We usually only need the first 29 bytes of the state response since they contain
+                # the current boolean parameters. The other parameters all have corresponding
+                # 'interrogate' commands.
+                num_bytes = 29
+            elif command[0] == 'I':
+                num_bytes = 14
+
+        # The first character is always a space or a leftover character from the previous command,
+        # when the number of bytes read was too large or too small.
+        bytes = self.adapter.read_bytes(num_bytes)[1:]
+        return bytes.decode('ascii').strip(' ,\r\n')
+
+    def values(self, command, separator=',', cast=float, preprocess_reply=None, **kwargs):
+        # I had to copy the values method from the adapter class since we need to call
```

Comment:
I don't think you need to reimplement `values`, since you already changed the `ask` method. Or I am missing something?

Reply:
See https://github.com/pymeasure/pymeasure/pull/559#discussion_r780658782

Reply:
I am sorry @CodingMarco, you are right and I missed your comment.
I will possibly file an issue such that this can be discussed/fixed in future.
One last point `ask` method is doing the strip so I am not sure it is also needed in `values`

Reply:
Yes you're right, I'll remove it in `values`.